source/common/stats/tag_extractor_impl.cc

```diff
@@ -114,13 +114,19 @@ bool TagExtractorStdRegexImpl::extractTag(TagExtractionContext& context, std::ve
                                                      regex_) &&
       match.size() > 1) {
     // remove_subexpr is the first submatch. It represents the portion of the string to be removed.
-    const auto& remove_subexpr = match[1];
+    uint32_t idx;
+    for (idx = 1; idx < match.size(); idx++) {
+      if (!match[idx].str().empty()) {
```

Comment:
just to be pedantic, how do we know that the empty string isn't the correct matched tag value?

I don't necessarily have a better solution but I'm just thinking about possible corner-cases with this.